src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/Publisher.java

```diff
@@ -18,33 +18,65 @@
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Action;
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.NonLeafNode;
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.scheduler.FlowUnitOperationArgWrapper;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class Publisher extends NonLeafNode<EmptyFlowUnit> {
 
   private static final Logger LOG = LogManager.getLogger(Publisher.class);
+  private final long initTime;
 
   private Collator collator;
   private boolean isMuted = false;
+  private Map<String, Long> actionToExecutionTime;
```

Comment:
Can we use Action itself as the key ?

Reply:
We could, but we'd have to implement hashCode() and equals() for all Actions. I asked Vigya, and he said it was a safe assumption that Action names are unique, so it should suffice as a key.